docs/Using-SRIOV-and-Clear-Containers.md

```diff
@@ -0,0 +1,278 @@
+
+   ```
+   $ cat /sys/bus/pci/devices/0000\:01\:00.0/sriov_totalvfs
+   63
+   $ cat /sys/bus/pci/devices/0000\:01\:00.1/sriov_totalvfs
+   63
+   ```
+   The previous commands show how many VFs you can create. The `sriov_totalvfs` 
+   file under sysfs for a PCI device specifies the total number of VFs that you
+   can create.
+
+4. Create the VFs:
+
+   ```
+   # echo 1 > /sys/bus/pci/devices/0000\:01\:00.0/sriov_numvfs
+   # echo 1 > /sys/bus/pci/devices/0000\:01\:00.1/sriov_numvfs
+   ```
+   
+   Create virtual functions by editing `sriov_numvfs`. In our example, we create
+   commands create virtual functions by editing `sriov_numvfs`. This example 
+   creates one VF per physical device. Note, creating one VF eliminates the 
+   usefulness of SR-IOV, and is done for simplicity in this example. 
+
+ 5. Verify the VFs were added to the host:
+
+   ```
+   # lspci | grep Ethernet | grep Virtual
+   02:10.0 Ethernet controller: Intel Corporation X540 Ethernet Controller Virtual Function (rev 01)
+   02:10.1 Ethernet controller: Intel Corporation X540 Ethernet Controller Virtual Function (rev 01)
+   ```
+
+6. Assign a MAC address to each VF:
+   
+   ```
+   $ ip link set <pf> vf <vfidx> mac <fake MAC address>
+   ```
+
+   Depending on the NIC being used, you  might need to explicitly set the MAC 
+   address for the VF device. Setting the MAC address guarantees that the 
+   address is consistent on the host and when passed to the guest. 
+
+## Example: Launch a Clear Container using SR-IOV
+
+The following example launches a Clear Containers container using SR-IOV:
+
+1. Build and start SR-IOV plugin:
+
+   This assumes you already have `GOPATH` set in your environment.
+   ```
+   $ sudo mkdir -p /etc/docker/plugins
+   $ sudo cp $GOPATH/src/github.com/clearcontainers/sriov/sriov.json /etc/docker/plugins/
+   $ cd $GOPATH/src/github.com/clearcontainers/sriov
+   $ go build
+   $ sudo ./sriov &
+   ```
+
+2. Create the docker network:
+
+   ```
+   $ sudo docker network create -d sriov --internal --opt pf_iface=enp1s0f0 --opt vlanid=100 --subnet=192.168.0.0/24 vfnet
+
+   E0505 09:35:40.550129    2541 plugin.go:297] Numvfs and Totalvfs are not same on the PF - Initialize numvfs to totalvfs
+   ee2e5a594f9e4d3796eda972f3b46e52342aea04cbae8e5eac9b2dd6ff37b067
+   ```
+   The previous commands create the SR-IOV docker network, a subnet, vlanid, 
+   and a physical interface, all which are required.
+  
+3. Start containers and test their connectivity
+
+   To start a container making use of SR-IOV:
+   ```
+   $ sudo docker run --runtime=cc-runtime --net=vfnet --ip=192.168.0.10 -it alpine
+   ```
+
+   If two machines with SR-IOV enabled NICs are connected back-to-back and each 
+   has a network with matching vlan-id created, use the following two commands 
+   to test the connectivity:
+   
+   Machine #1:
+   ```
+   sriov@sriov-2:~$ sudo docker run --runtime=cc-runtime --net=vfnet --ip=192.168.0.10 -it mcastelino/iperf iperf3 -s
```

Comment:
Nits:

- If this is "Machine 1", you'd expect the prompt to be `sriov@sriov-1:~$`, not `sriov@sriov-2:~$`.
- These prompts are a bit complex - wouldn't, `sriov-1:$ ` and `sriov-2:$ ` suffice?

Reply:
fixed